File: 278.py
# The isBadVersion API is already defined for you.
# @param version, an integer
# @return a bool
# def isBadVersion(version):

class Solution:
    def firstBadVersion(self, n):
        """
        :type n: int
        :rtype: int
        """
        a=1
        b=n
        while (a<b):
            c=(a+b)//2
            if isBadVersion(c):
                b=c
            else:
                a=c+1
        return a

File: 278.cpp
// Forward declaration of isBadVersion API.
bool isBadVersion(int version);

class Solution {
public:
    int firstBadVersion(int n) {
        long long a=1,b=n;
        while (a<b)
        {
            long long c=(a+b)/2;
            if (isBadVersion(c))
                b=c;
            else
                a=c+1;
        }
        return a;
    }
};
